src/ui/views/Approval/components/TxComponents/GasSelecter.tsx

```diff
@@ -216,11 +196,74 @@ const GasSelector = ({
       setCustomNonce(Number(e.target.value));
     }
   };
-
-  const handleClickAdvance = () => {
-    setAdvanceExpanded(!advanceExpanded);
+  const getLastTimeSavedGas = async () => {
+    const {
+      gasPrice = null,
+      gasLevel = null,
+      lastTimeSelect = null,
+    } = await wallet.getLastTimeGasSelection(chainId);
+    if (gasPrice) {
+      setCustomGas(gasPrice);
+    }
+    if (lastTimeSelect === 'gasLevel') {
+      const lastSected = gasList.filter((item) => item.level === gasLevel);
```

Comment:
typo, should be lastSelected?

Reply:
btw Array.filter works but not best, when we just want to find one correct item, Array.find should be better especially in a large Array since Array.find will stop loop when find the first one matched

Reply:
done